### PR TITLE
Fix arbitrary function for structural type

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

**Comment by martint** [comment]
Explain what's the bug in the commit message. Something like "there was a bug when XXX that caused YYY to happen"

**Comment by haozhun** [comment]
@martint The bug is `arbitrary` function does not work for structural type

**Comment by martint** [comment]
How about "Invoking arbitrary() over a structural type caused XXX error due to YYY". The idea is that someone can look at the commit and 1) be able to tell whether this fix applies to something they may be seeing 2) understand the reason why it's broken 3) have some context when looking at the actual changes in the commit (trying to reverse-engineer the observed behavior from the code changes can be very hard)

**Comment by haozhun** [comment]
Updated commit message.

**Comment by cberner** [comment]
Lets rewrite `arbitrary()` to be similar to `AbstractMinMax`. Using an array as the serialized type will incur additional cost, and also switching to not use `getSingleValueBlock` would be a big performance win. (I tested a query with `max()` instead of `arbitrary()` and it was much faster)

**Comment by haozhun** [comment]
Rewritten as a 2nd commit

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArbitraryAggregation.java [hunk 1]
```diff
@@ -70,32 +88,67 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
         return new FunctionInfo(signature, getDescription(), aggregation);
     }
 
-    private static InternalAggregationFunction generateAggregation(Type valueType)
+    protected InternalAggregationFunction generateAggregation(Type type)
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArbitraryAggregation.java]: change this back to static?

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArbitraryAggregation.java [hunk 2]
```diff
@@ -70,32 +88,67 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
...
     {
-        DynamicClassLoader classLoader = new DynamicClassLoader(ArbitraryAggregation.class.getClassLoader());
+        DynamicClassLoader classLoader = new DynamicClassLoader(AbstractMinMaxAggregation.class.getClassLoader());
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArbitraryAggregation.java]: wrong classloader I think

**Comment by cberner** [comment]
couple comments. otherwise looks good

**Pull request merged**